autonomi/src/client/data_types/chunk.rs

```diff
@@ -238,22 +238,38 @@ impl Client {
 
         loop {
             let mut upload_tasks = vec![];
-            for chunk in chunks {
+            let total_chunks = chunks.len();
+            for (i, &chunk) in chunks.iter().enumerate() {
                 let self_clone = self.clone();
                 let address = *chunk.address();
 
                 let Some((proof, _)) = receipt.get(chunk.name()) else {
                     debug!("Chunk at {address:?} was already paid for so skipping");
+                    #[cfg(feature = "loud")]
+                    println!(
+                        "({}/{}) Chunk stored at: {} (skipping, already exists)",
+                        i + 1,
+                        chunks.len(),
+                        chunk.address().to_hex()
+                    );
                     continue;
                 };
 
                 upload_tasks.push(async move {
-                    self_clone
+                    let res = self_clone
                         .chunk_upload_with_payment(chunk, proof.clone())
                         .await
                         .inspect_err(|err| error!("Error uploading chunk {address:?} :{err:?}"))
                         // Return chunk reference too, to re-use it next attempt/iteration
-                        .map_err(|err| (chunk, err))
+                        .map_err(|err| (chunk, err));
+                    #[cfg(feature = "loud")]
+                    println!(
+                        "({}/{}) Chunk stored at: {}",
+                        i + 1,
```

Comment:
Here I would reword this to 
```diff
- "({}/{}) Chunk stored at: {}",
+ "({}/{}) Uploading Chunk at: {}",
```
As we did not store it yet at time of printing!

Reply:
If I understand correctly, `put_record` waits for it to be verified, this is call by `chunk_upload_with_payment`. Which also includes this line:
```
debug!("Successfully stored chunk: {chunk:?} to {storing_nodes:?}");
```

Though now that I look at it, the added `println!` statement should only be executed if `res.is_ok()`.